bobcat/bb-core/src/main/java/com/cognifide/qa/bb/utils/PropertyUtils.java

```diff
@@ -0,0 +1,138 @@
+/*-
+ * #%L
+ * Bobcat
+ * %%
+ * Copyright (C) 2016 Cognifide Ltd.
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+package com.cognifide.qa.bb.utils;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.nio.charset.StandardCharsets;
+import java.util.Properties;
+
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.cognifide.qa.bb.PropertyBinder;
+import com.cognifide.qa.bb.constants.ConfigKeys;
+import com.cognifide.qa.bb.constants.Timeouts;
+
+public class PropertyUtils {
+
+	private static final Logger LOG = LoggerFactory.getLogger(PropertyUtils.class);
+
+	private static final String[] PREFIXES = new String[] { "webdriver.", "phantomjs." };
+
+	private PropertyUtils() {
+		// util class...
+	}
+
+	/**
+	 * This util method gather system and class path properties, and returns them as <code>Properties</code>
+	 * object.
+	 * 
+	 * @return Properties - object with all properties.
+	 */
+	public static Properties gatherProperties() {
```

Comment:
couldn't this be cached somehow? you are using this method in Bobcumber class which is loading these prop from the scratch